src/contract/basic_node_signer.ts

```diff
@@ -18,10 +18,16 @@ export const basicNodeSigner = (
   networkPassphrase: string,
 ) => ({
   // eslint-disable-next-line require-await
-  signTransaction: async (tx: string) => {
-    const t = TransactionBuilder.fromXDR(tx, networkPassphrase);
-    t.sign(keypair);
-    return t.toXDR();
+  signTransaction: async (tx: string, signer?, opts?: {
```

Comment:
`signer?,` is not valid TypeScript; does this code work?

I think this whole approach will cause typing problems, though! Sorry I didn't catch this before you started working. See comment here:

- https://github.com/stellar/js-stellar-sdk/issues/1063#issuecomment-2430148144

Reply:
Ok, conversation resolved, this is the desired way to do it. Please make sure your code works by testing it and we can get this merged!